### PR TITLE
Address ship review feedback

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/0/1207977044044823/f

### Description
See attached task description

### Steps to test this PR
UI changes have been approved by design. Smoke test if needed each item in description if needed.

### CODE REVIEW

**Comment by karlenDimla** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4864?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4864** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4864?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4855** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4855?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4849** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4849?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4840** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4840?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karlenDimla and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 1]
```diff
@@ -86,21 +88,44 @@ class SubscriptionFeedbackViewModel @Inject constructor(
     fun onReportTypeSelected(reportType: SubscriptionFeedbackReportType) {
         viewModelScope.launch {
             val previousFragmentState = viewState.value.currentFragmentState
-            val newMetadata = viewState.value.feedbackMetadata.copy(
+            var newMetadata = viewState.value.feedbackMetadata.copy(
                 reportType = reportType,
             )
```
Comment (bemusementpark) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:93]: Would perhaps be easier to read, and more functional paradigm to keep it a `val` and fully construct here:

```suggestion
            val newMetadata = viewState.value.feedbackMetadata.let {
                it.copy(
                    reportType = reportType,
                    category = when (it.source) {
                        SUBSCRIPTION_SETTINGS -> SUBS_AND_PAYMENTS
                        VPN_MANAGEMENT, VPN_EXCLUDED_APPS -> VPN
                        else -> it.category
                    }
                )
            }
```

Then just use this `newMetadata` as normal down below, and we won't have to let it be reassigned as a side-effect of creating `nextState`.

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 2]
```diff
@@ -86,21 +88,44 @@ class SubscriptionFeedbackViewModel @Inject constructor(
...
+
+                        else -> {
+                            FeedbackCategory(reportType.asTitle())
+                        }
+                    }
+                }
+
+                GENERAL_FEEDBACK -> FeedbackSubmit(reportType.asTitle())
+                REQUEST_FEATURE -> FeedbackSubmit(reportType.asTitle())
+            }
```
Comment (bemusementpark) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:117]:
```suggestion
            val nextState = when (reportType) {
                REPORT_PROBLEM -> {
                    when (newMetadata.source) {
                        SUBSCRIPTION_SETTINGS, VPN_MANAGEMENT, VPN_EXCLUDED_APPS -> {
                            FeedbackSubCategory(newMetadata.category!!.asTitle())
                        }
                        else -> FeedbackCategory(reportType.asTitle())
                    }
                }

                GENERAL_FEEDBACK, REQUEST_FEATURE -> FeedbackSubmit(reportType.asTitle())
            }
```

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 3]
```diff
@@ -305,14 +328,28 @@ class SubscriptionFeedbackViewModel @Inject constructor(
...
-                    )
+                    is FeedbackSubCategory -> {
+                        val previousState =
+                            if (currentFeedbackMetadata.reportType == REPORT_PROBLEM &&
+                                (
+                                    currentFeedbackMetadata.source == SUBSCRIPTION_SETTINGS ||
+                                        currentFeedbackMetadata.source == VPN_MANAGEMENT ||
+                                        currentFeedbackMetadata.source == VPN_EXCLUDED_APPS
+                                    )
+                            ) {
```
Comment (bemusementpark) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:339]: Might be nice to extract to courtesy (possibly extension) property or function for readability. e.g.

```suggestion
                            if (currentFeedbackMetadata.isReportAboutSubscriptionOrVpn) {
```

**Review by bemusementpark — COMMENTED** [review]
some ideas

**Review by marcosholgado — APPROVED** [review]
LGTM

**Pull request merged**